src/parcsr_ls/par_amg.c

```diff
@@ -3576,8 +3598,19 @@ hypre_BoomerAMGSetDofFunc( void     *data,
       hypre_error_in_arg(1);
       return hypre_error_flag;
    }
-   hypre_TFree(hypre_ParAMGDataDofFunc(amg_data), HYPRE_MEMORY_HOST);
-   hypre_ParAMGDataDofFunc(amg_data) = dof_func;
+   hypre_IntArrayDestroy(hypre_ParAMGDataDofFunc(amg_data));
+   /* NOTE: size of hypre_IntArray will be set during AMG setup 
+    *       the memory location is assumed to be host */
+   if (dof_func == NULL)
+   {
+      hypre_ParAMGDataDofFunc(amg_data) = NULL;
+   }
+   else
+   {
+      hypre_ParAMGDataDofFunc(amg_data) = hypre_IntArrayCreate(-1);
+      hypre_IntArrayMemoryLocation(hypre_ParAMGDataDofFunc(amg_data)) = HYPRE_MEMORY_HOST;
```

Comment:
Do we assume `dof_func` is always a host pointer?

Reply:
Per our discussion last week, I will change this to assume a device pointer.